zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -1411,14 +1440,46 @@ impl<F: Field> ExecutionConfig<F> {
                 rev_count,
                 step.reversible_write_counter_delta
             );
+
             // In the EVM Circuit, reversion rw lookups are assigned after their
             // corresponding rw lookup, but in the bus-mapping they are
             // generated at the end of the step.
             let idx = if is_rev {
                 step.rw_indices_len() - rev_count
             } else {
-                idx - rev_count
+                idx - rev_count + offset - copy_lookup_processed as usize
```

Comment:
Here copy_lookup_included looks pretty special: it's a boolean flag and will only toggle from false -> true when suffer first copy event. Just wanna double confirm if there are more than 1 copy event does the idx calculation still correct~?

Reply:
yes, it only supports 1 copy event and will add a `TODO` and warning message as we discussed.
https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1482#issuecomment-1600378826